lib/ansible/modules/database/postgresql/postgresql_user.py

```diff
@@ -244,6 +257,110 @@ class InvalidFlagsError(Exception):
 class InvalidPrivsError(Exception):
     pass
 
+
+class NoSaslPrepError(Exception):
+    pass
+
+# ===========================================
+# Helper utility functions
+#
+
+
+"""An implementation of RFC4013 SASLprep.
+This code is mostly (apart from utils methods) a copy of
+https://github.com/mongodb/mongo-python-driver/blob/master/pymongo/saslprep.py
+"""
```

Comment:
I think that licenses for GPLv3+ and Apachev2 allow us to mix this code legally but I'm a little hesitant to say this is the right way to include the code.  It has a separate upstream and a separate license so maybe it would be better to but this into its own file in module_utils and then import it into this code?

That would also make it theoretically possible for us to track updates in the upstream code (in case there were security issues or bugs that we'd want to address, for instance)

Reply:
@abadger, I'm not really good at understanding of Ansible internals for a while but code separating sounds reasonably. Besides, in the future, it can be useful for other modules. @astorath can you realized it in own file as a class? (Better in a general reusable form)

Reply:
@Andersson007 sorry, I'm in a business trip right now, so I'll finish next week.